src/javascript/app/pages/user/view_popup/view_popup.js

```diff
@@ -553,30 +554,47 @@ const ViewPopup = (() => {
-                appendAuditLink('trade_details_entry_spot');
-            } else {
-                contract_starts.div.remove();
-            }
-            // don't show exit tick information if missing or manual sold
-            if (contract.audit_details.contract_end && contract.status !== 'sold') {
-                const contract_ends = createAuditTable(localize('Contract Ends'));
-                parseAuditResponse(contract_ends.table, contract.audit_details.contract_end).then(() => {
-                    if (contract.audit_details.contract_end) {
-                        createAuditHeader(contract_ends.table);
-                        appendAuditLink('trade_details_current_spot');
-                    } else {
-                        contract_ends.div.remove();
-                    }
+        if (!contract.tick_count) {
+            const contract_starts = createAuditTable(localize('Contract Starts'));
+            parseAuditResponse(contract_starts.table, contract.audit_details.contract_start).then(() => {
+                if (contract.audit_details.contract_start) {
+                    createAuditHeader(contract_starts.table);
+                    appendAuditLink('trade_details_entry_spot');
+                } else {
+                    contract_starts.div.remove();
+                }
+                // don't show exit tick information if missing or manual sold
+                if (contract.audit_details.contract_end && contract.status !== 'sold') {
+                    const contract_ends = createAuditTable(localize('Contract Ends'));
+                    parseAuditResponse(contract_ends.table, contract.audit_details.contract_end).then(() => {
+                        if (contract.audit_details.contract_end) {
+                            createAuditHeader(contract_ends.table);
+                            appendAuditLink('trade_details_current_spot');
+                        } else {
+                            contract_ends.div.remove();
+                        }
+                        onAuditTableComplete(show_audit_table);
+                    });
+                } else {
                     onAuditTableComplete(show_audit_table);
-                });
-            } else {
-                onAuditTableComplete(show_audit_table);
-            }
-        });
+                }
+            });
+        } else {
+            const contract_details = createAuditTable(localize('Contract Details'));
+            parseAuditResponse(contract_details.table, contract.audit_details.all_ticks).then(() => {
+                if (contract.audit_details.all_ticks) {
+                    createAuditHeader(contract_details.table);
+                    appendAuditLink('trade_details_entry_spot');
+                    appendAuditLink('trade_details_current_spot');
+                } else {
+                    contract_details.div.remove();
+                }
+
+                if (contract.status === 'sold') {
```

Comment:
is this the best way to check that contract has ended? why not check contract.status !== 'open'?

Reply:
yes, you are correct. its better to check if the contract.status is not open 👍